### PR TITLE
Added various githooks

### PR DESCRIPTION
**Pull request by awolkers**
After `npm i` has finished the `gulp githooks` task will be executed. This task copies all the available [git hooks](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks) from `/githooks/` to the `/.git/hooks/` folder. The following hooks are added by default.

1. **post-checkout** This hook will check if changes are made to package.json. If so `npm prune` and `npm i` will be execuded. This hook also checks if changes are made to the `githooks/` folder, if so `gulp githooks` will be executed.
2. **post-merge** This hooks works the same as the **post-checkout** hook.
3. **pre-commit** This hooks checks if changes where made to any `.scss` of `.js` files, if so then execude the related linting task.
4. **prepare-commit-msg** This hook will prepends the branch name to the commit message if the current branch is any branch but master

Fixed a small url related typo to one of contributors within `package.json`.  
Changed `js-test` to `js-lint` which is semantically better.  
Added `css-lint` task which currenlty does nothing.

### CODE REVIEW

File: tasks/githooks.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+var config       = require('../config');
+var gulp         = require('gulp');
+var del          = require('del');
```
Comment (branneman) [inline tasks/githooks.js]: So many spaces?

File: gulpfile.js [hunk 2]
```diff
@@ -22,13 +23,18 @@ gulp.task('img-watch', ['img-optimize'], tasksImages.watch);
...
 
+// Githook tasks
+gulp.task('githooks-clean', tasksGithooks.clean);
+gulp.task('githooks', ['githooks-clean'], tasksGithooks.copy);
+
 // Group-tasks
 gulp.task('dev', ['browsersync', 'html-watch', 'img-watch', 'css-watch', 'js-watch']);
 gulp.task('dist', [/*'clean', */'html-compile', 'img-optimize', 'css-compile', 'js-transpile']);
-gulp.task('test', ['js-test']);
+gulp.task('test', ['js-lint']);
```
Comment (branneman) [inline gulpfile.js]: I think the `test` task should also run the `css-lint` task.

File: package.json [hunk 3]
```diff
@@ -7,12 +7,13 @@
         "url": "https://github.com/mirabeau-nl/frontend-bootstrap.git"
     },
     "scripts": {
+        "postinstall": "gulp githooks",
```
Comment (branneman) [inline package.json]: Let's not introduce a global gulp dependency:
```
"node_modules/gulp/bin/gulp.js githooks"
```

**Comment by branneman** [comment]
image

It seems the exit code of the `js-lint` task is 0, which means there's no fail, where it should have given something different than zero. This means a `git commit` is still executed, instead of being blocked.

**Comment by awolkers** [comment]
Added error reporters for jshint and jscs tasks so commit will now fail on error

**Pull request merged**